### PR TITLE
Improve setup.py and add dependency check

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR makes sure that when PEFT is installed but it doesn't match what's written in the setup.py it fails gracefully. 

The following will fail:
```
!pip install git+https://github.com/huggingface/peft.git
!pip install diffusers 
```
and 
```py
import diffusers
```

with:
```
ImportError: peft<=0.6.2 is required for a normal functioning of this module, but found peft==0.6.3.dev0.
```

### CODE REVIEW

File: src/diffusers/utils/versions.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Copyright 2020 The HuggingFace Team. All rights reserved.
```
Comment (patrickvonplaten) [inline src/diffusers/utils/versions.py:1]: Copied from `transformers`

**Comment by BenjaminBossan** [comment]
Thanks for adding the check. There probably needs to be an option to toggle this check off so that we can run CI on `main` PEFT and discover potential issues early. Perhaps via some environment variable, could be "private" and only for our testing.

**Comment by sayakpaul** [comment]
> There probably needs to be an option to toggle this check off so that we can run CI on main PEFT and discover potential issues early. Perhaps via some environment variable, could be "private" and only for our testing.

Tend to agree with this. Otherwise, we won't be able to catch issues. What's the plan for that, @patrickvonplaten?

File: src/diffusers/utils/versions.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    "<": operator.lt,
+    "<=": operator.le,
+    "==": operator.eq,
+    "!=": operator.ne,
+    ">=": operator.ge,
+    ">": operator.gt,
+}
+
+
+def _compare_versions(op, got_ver, want_ver, requirement, pkg, hint):
```
Comment (sayakpaul) [inline src/diffusers/utils/versions.py:37]: Nice. Then we should totally use it remove functions like `compare_torch_versions()`.

**Comment by patrickvonplaten** [comment]
> Thanks for adding the check. There probably needs to be an option to toggle this check off so that we can run CI on `main` PEFT and discover potential issues early. Perhaps via some environment variable, could be "private" and only for our testing.

Good point!

**Comment by patrickvonplaten** [comment]
New functionality if PEFT > 0.6.2 is installed:

```
python -c "import diffusers"
```
will fail

```
CHECK_PEFT=0 python -c "import diffusers"
```

won't fail

**Comment by sayakpaul** [comment]
That works for me!

**Pull request merged**